### ISSUE TITLE
Update to support Minecraft 1.21

### ISSUE THREAD

**Issue by mettlephaontoms** [Siphalor/amecs#87]
My playstyle is dependent on AMECS for managing keybinds, My abilities are reduced as long as I'm without this mod, please update to 1.21.

**Comment by FyrGlow** [comment]
Yeah, I'm looking forward to it, too

**Comment by MeeniMc** [comment]
The 1.20.4 version works on 1.21 for me

**Comment by XavierBit** [comment]
There is a 1.21 fork here: https://github.com/wyatt-herkamp/too-many-shortcuts

**Comment by wilebaldo** [comment]
> The 1.20.4 version works on 1.21 for me

How is working the same version as the 1.20.4 on 1.21? When I copy the mods in Minecraft that is not built it for the specific version I'm playing is automatic crash the game.

**Comment by MeeniMc** [comment]
Taking it back, it doesn't work by itself, but I have another (as of yet unidentified) mod that changes the accesswidener on the same target class and makes it work. Probably means that the only problem to run it is a missing accesswidener.

**Comment by SirPotato42** [comment]
> Taking it back, it doesn't work by itself, but I have another (as of yet unidentified) mod that changes the accesswidener on the same target class and makes it work. Probably means that the only problem to run it is a missing accesswidener.

Did you find the mod?

**Comment by xxxkingxl** [comment]
https://modrinth.com/mod/accesswidener-loader
I found it
 I think

**Comment by xxxkingxl** [comment]
I am trying it, because I want to use this mod in my modpack as soon as possible
the alternative is not that great at all

**Comment by SirPotato42** [comment]
> I am trying it, because I want to use this mod in my modpack as soon as possible the alternative is not that great at all

Yeah, I tried the link you sent (https://modrinth.com/mod/accesswidener-loader), but it was not very good
I am using the controlling mod in the meantime, but I would definitely love to see a 1.21 version for amecs soon

**Comment by SirPotato42** [comment]
> I am trying it, because I want to use this mod in my modpack as soon as possible the alternative is not that great at all

Yeah, I tried the link you sent (https://modrinth.com/mod/accesswidener-loader), but it was not very good
I am using the controlling mod in the meantime, but I would definitely love to see a 1.21 version for amecs soon

**Comment by wilebaldo** [comment]
> > Taking it back, it doesn't work by itself, but I have another (as of yet unidentified) mod that changes the accesswidener on the same target class and makes it work. Probably means that the only problem to run it is a missing accesswidener.
> 
> Did you find the mod?

I'm found one is working almost the same, the name is "Too Many Shortcuts", also when you have many Keybinding you can use this other mod also is called "Too Many Binds".

**Comment by MeeniMc** [comment]
The mod that makes amecs work on 1.21+ is **HorseBuff**, yep, that is completely random.

**Comment by NeonExcel** [comment]
Has anyone found a alternative?

**Comment by wilebaldo** [comment]
> Has anyone found a alternative?

Yes, this is the alternative I'm using with the same configuration without a problem.

> There is a 1.21 fork here: https://github.com/wyatt-herkamp/too-many-shortcuts

**Comment by notrohit1702** [comment]
> > Has anyone found a alternative?
> 
> Yes, this is the alternative I'm using with the same configuration without a problem.
> 
> > There is a 1.21 fork here: https://github.com/wyatt-herkamp/too-many-shortcuts

Is there no better option, that mod crashed when I first tried it and feels a bit low-effort

**Comment by wilebaldo** [comment]
> Is there no better option, that mod crashed when I first tried it and feels a bit low-effort

Really I don't know, this is the mod I'm using and is working fine for me. Check do you download the correct version for your version of minecraft. I'm playing on Minecraft 1.21 and the version for the mod is 0.0.5 and is working.

**Comment by NeonExcel** [comment]
> > Has anyone found a alternative?
> 
> Yes, this is the alternative I'm using with the same configuration without a problem.
> 
> > There is a 1.21 fork here: https://github.com/wyatt-herkamp/too-many-shortcuts

I tried this mod, And all the keybinds i set work until i load into a world. Then it just reverts to defaults.

**Comment by B14CK313** [comment]
While searching for a replacement, I found this mod (recently updated to 1.21.4):
https://www.curseforge.com/minecraft/mc-mods/modern-keybinding-fabric

Working great so far. Hope it helps somebody!

**Comment by MisterCheezeCake** [comment]
I have ported Amecs to 1.21 and added some new features including long requested Controlling support. You can find the mod here:

https://modrinth.com/mod/amecs-reborn

**Comment by Siphalor** [comment]
Hi, I just noticed this issue and am happy to say that I added support for Minecraft 1.21 last fall in version 1.4.0.
